ci-operator/step-registry/ibmcloud/deprovision/vpc/ibmcloud-deprovision-vpc-commands.sh

```diff
@@ -105,8 +105,14 @@ echo "Using region: ${region}  resource_group: ${resource_group} vpc: ${vpc_name
 
 "${IBMCLOUD_CLI}" target -g ${resource_group} -r ${region}
 
-echo "DEBUG" "Removing the vpc ${vpc_name}"
+echo "DEBUG" "Removing the vpc ${vpc_name} ..."
 delete_vpc "${vpc_name}"
 
+echo "DEBUG" "Removing the resource reclamations ..."
+hasReclamations=$("${IBMCLOUD_CLI}" resource reclamations |  awk '(NR==4) {print $1}')
+if [ "${hasReclamations}" != "No" ]; then
+  ${IBMCLOUD_CLI} resource reclamations |  awk '(NR>4) {print $1}' | xargs -n1 ibmcloud resource reclamation-delete -f
```

Comment:
My points is not deleting the reclamations resources irrelevant with the installed cluster, or even we can keep the issue remaining, but run the commands locally.

Reply:
the reclamations will block cleanup the resource group, then the step will failed.

Reply:
Let me confirm, will the command will delete all the reclamations, even not be created from this cluster install, right?

Reply:
will delete the reclamations under the resource group (the vpc created under this group)

Reply:
If that, I think you do not need the line to check:
```
hasReclamations=$("${IBMCLOUD_CLI}" resource reclamations | awk '(NR==4) {print $1}')
```
And I think the removal of reclamations should happen prior to `delete_vpc`.

Reply:
why the removal of reclamations should happen prior to delete_vpc?

Reply:
Per your pasted https://issues.redhat.com/browse/OCPBUGS-7511, from the error log:
```
time="2023-02-13T10:40:54Z" level=debug msg="Failed to delete resource group ci-op-t87h9imn-5767e-s89bj: Resource groups with active or pending reclamation instances can't be deleted. Use the CLI commands \"ibmcloud resource service-instances --type all\" and \"ibmcloud resource reclamations\" to check for remaining instances, then delete the instances and try again."
```

Because of active or pending reclamation instances, that lead the destroy failed, but per my understanding, instances are created inside vpc, so vpc has dependency of instances, so need to ensure no dependencies before  deleting vpc.

Reply:
checked the IBM Doc [Using resource reclamations](https://cloud.ibm.com/docs/account?topic=account-resource-reclamation&interface=cli) , I need to clean up all the reclaimed resources of the group. even the vpc maybe has a reclamation.

Reply:
You can run `ibmcloud resource reclamation-delete` twice before and after `delete_vpc`

Reply:
[OCPBUGS-7511](https://issues.redhat.com/browse/OCPBUGS-7511),  is an image of the cluster failing to be deleted. Now in ibmcloud-deprovision-vpc step ,  I add the 'ibmcloud resource reclamation-delete' before destroying the resource group (used for creating the vpc), why need more 'ibmcloud resource reclamation-delete' before delete_vpc ?

Reply:
I do not know the accurate details of OCPBUGS-7511, from the error log, I thought it was caused by remaining instances, we can not foresee which resources would be remaining in the future, if the remaining resource are instances, we need to clean up reclamation before `delete_vpc`.

Whatever, if you insist with one place of reclamation-delete according to your only known case in OCPBUGS-7511, I am okay with it.  Just like what I mentioned in the beginning of this thread, I even think we should never add the workaround in the script, I do not have preference of `reclamation-delete`.

Reply:
checked the test history 
 [reclamation-delete succeed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/36512/rehearse-36512-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-ibmcloud/1627675502225395712/artifacts/installer-rehearse-ibmcloud/ibmcloud-deprovision-vpc/build-log.txt)
without reclamation-delete 
- [delete resource group failed](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/36512/rehearse-36512-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-ibmcloud-ipi-private-fips-p2-f7/1627516679871270912/artifacts/ibmcloud-ipi-private-fips-p2-f7/ibmcloud-deprovision-vpc/build-log.txt)
- [delete resource group succeed](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/36512/rehearse-36512-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-ibmcloud-ipi-private-fips-p2-f7/1626641822568681472/artifacts/ibmcloud-ipi-private-fips-p2-f7/ibmcloud-deprovision-vpc/build-log.txt )

Reply:
Let us stop discussion to add how many reclamation-delete commands here, the only comment now is https://github.com/openshift/release/pull/36512#discussion_r1112731435

Reply:
when no reclamations, the reclamation-delete will caused error. 
$ ic resource reclamations | awk '(NR==4) {print $1}'
No
$ ic resource reclamations | awk '(NR==4) {print $1}'| xargs -n1 ibmcloud resource reclamation-delete -f
Submitting request to delete resource reclamation No under account fdc2e14cf8bc4d53a67f972dc2e2c861 as ServiceId-dff277a9-b608-410a-ad24-c544e59e3778...
FAILED
Resource reclamation 'No' was not found.

Reply:
Can you show me an example output when reclamations are existing?

Reply:
```
 ic resource reclamations | awk '(NR==4) {print $1}'| xargs -n1 ibmcloud resource reclamation-delete -f
```
Do you means only the resources shown on 4th line will be deleted?
If there are multiple lines for reclamations resources, how to deal with?

Reply:
To easy reading for the code, I think it should be:
```
echo "DEBUG" "Removing the resource reclamations ..."
if [[ $(ibmcloud resource reclamations -q) == "No reclamation found" ]]; then
  echo "No reclamation found"
else
  ${IBMCLOUD_CLI} resource reclamations -q |  awk '{print $1}' | xargs -n1 ibmcloud resource reclamation-delete -f
fi
```